spring-cloud-aws-parameter-store-config/src/main/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertySourceLocator.java

```diff
@@ -73,56 +73,20 @@ public AwsParamStorePropertySourceLocator(AWSSimpleSystemsManagement ssmClient,
 
 		ConfigurableEnvironment env = (ConfigurableEnvironment) environment;
 
-		String appName = properties.getName();
-
-		if (appName == null) {
-			appName = env.getProperty("spring.application.name");
-		}
+		AwsParamStorePropertySources sources = new AwsParamStorePropertySources(this.properties, this.logger);
 
 		List<String> profiles = Arrays.asList(env.getActiveProfiles());
-
-		String prefix = this.properties.getPrefix();
-
-		String appContext = prefix + "/" + appName;
-		addProfiles(this.contexts, appContext, profiles);
-		this.contexts.add(appContext + "/");
-
-		String defaultContext = prefix + "/" + this.properties.getDefaultContext();
-		addProfiles(this.contexts, defaultContext, profiles);
-		this.contexts.add(defaultContext + "/");
+		this.contexts.addAll(sources.getAutomaticContexts(profiles));
 
 		CompositePropertySource composite = new CompositePropertySource("aws-param-store");
 
 		for (String propertySourceContext : this.contexts) {
-			try {
-				composite.addPropertySource(create(propertySourceContext));
-			}
-			catch (Exception e) {
-				if (this.properties.isFailFast()) {
-					logger.error(
-							"Fail fast is set and there was an error reading configuration from AWS Parameter Store:\n"
-									+ e.getMessage());
-					ReflectionUtils.rethrowRuntimeException(e);
-				}
-				else {
-					logger.warn("Unable to load AWS config from " + propertySourceContext, e);
-				}
-			}
+			PropertySource<AWSSimpleSystemsManagement> propertySource = sources
+					.createPropertySource(propertySourceContext, true, this.ssmClient);
+			composite.addPropertySource(propertySource);
```

Comment:
Since `createPropertySource` can return `null`, it should be added to composite only if it's not null.